packages/twenty-front/src/modules/object-record/hooks/useFindDuplicateRecords.ts

```diff
@@ -38,46 +38,48 @@ export const useFindDuplicateRecords = <T extends ObjectRecord = ObjectRecord>({
     objectMetadataItem.nameSingular,
   );
 
-  const { data, loading, error } = useQuery<RecordGqlOperationFindManyResult>(
-    findDuplicateRecordsQuery,
-    {
-      variables: {
-        id: objectRecordId,
+  const { data, loading, error } =
+    useQuery<RecordGqlOperationFindDuplicatesResult>(
+      findDuplicateRecordsQuery,
+      {
+        variables: {
+          ids: objectRecordIds,
+        },
+        onCompleted: (data) => {
+          onCompleted?.(data[queryResponseField]);
+        },
+        onError: (error) => {
+          logError(
+            `useFindDuplicateRecords for "${objectMetadataItem.nameSingular}" error : ` +
+              error,
+          );
+          enqueueSnackBar(
+            `Error during useFindDuplicateRecords for "${objectMetadataItem.nameSingular}", ${error.message}`,
+            {
+              variant: SnackBarVariant.Error,
+            },
+          );
+        },
       },
-      onCompleted: (data) => {
-        onCompleted?.(data[queryResponseField]);
-      },
-      onError: (error) => {
-        logError(
-          `useFindDuplicateRecords for "${objectMetadataItem.nameSingular}" error : ` +
-            error,
-        );
-        enqueueSnackBar(
-          `Error during useFindDuplicateRecords for "${objectMetadataItem.nameSingular}", ${error.message}`,
-          {
-            variant: SnackBarVariant.Error,
-          },
-        );
-      },
-    },
-  );
+    );
 
-  const objectRecordConnection = data?.[queryResponseField];
+  const objectResults = data?.[queryResponseField];
 
-  const records = useMemo(
+  const results = useMemo(
     () =>
-      objectRecordConnection
-        ? (getRecordsFromRecordConnection({
-            recordConnection: objectRecordConnection,
-          }) as T[])
-        : [],
-    [objectRecordConnection],
+      objectResults?.map((result: RecordGqlConnection) => {
+        return result
+          ? (getRecordsFromRecordConnection({
+              recordConnection: result,
+            }) as T[])
+          : [];
+      }),
+    [objectResults],
   );
 
   return {
     objectMetadataItem,
-    records,
-    totalCount: objectRecordConnection?.totalCount,
```

Comment:
What happened to totalCount? 😄 We don't need it anymore?

Reply:
It's within the query itself, the count here would now be equal to the number of ids you sent, I guess if you ask for totalCount as a field within the query fields it should work but I didn't try :-/